src/utils/matter-parser.js

```diff
@@ -0,0 +1,98 @@
+const yaml = require('yaml');
+const fsp = require('fs').promises;
+const glob = require('glob');
+const { promisify } = require('util');
+
+/**
+ * Class for parsing a markdown file into frontmatter and content
+ */
+class MatterParser {
+  /**
```

Comment:
Interesting that you chose a class to represent this process. I think the static functions point that this could be a datatype + utility functions. This is totally fine though, just offering an alternative arch. 💁🏻